### PR TITLE
Add a review app for the components

### PR DESCRIPTION
**Pull request by aliuk2012**
I'd like to apologize for this complex PR to add a review app for us to be able to view and test our components in isolation. The review application is mainly based off govuk-frontend. The review app is far from being perfect but its a start in the right direction for us. None of the changes made in this PR should affect the package which we end up using in our various applications.

# Test
- Pull down the branch
- `npm install`
- `npm run dev` it should open your browser and load up localhost:3001
- You should be able to see an example of the footer component.

Further improvements will be made to the review app in further PR's such as adding compiled JavaScript etc. but in a tempt to keep this Pr as small as possible. I  have done just enough to get a basic working example.

Ticket: https://trello.com/c/FE2NINme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,45 @@ To be notified when there’s a new release:
...
+
+## Developing new components
+
+We use gulp to automate various tasks. Below we will detail what each part of the process is
+and what task are run
+
+
+## 1. Installing the repo
+After cloning this repository you will need to run `npm install`. After npm has successfully installed
+all packages and dependencies, an npm `postinstall` script will automatically run to clea
```
Comment (benvand) [inline README.md]: `to clea` typo? :)

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  // Turn form POSTs into data that can be used for validation.
+  app.use(bodyParser.urlencoded({ extended: true }))
+
+  // Handle the banner component serverside.
+  require('./banner.js')(app)
+
+  // Define routes
+
+  // Index page - render the component list template
+  app.get('/', async function (req, res) {
```
Comment (benvand) [inline app/app.js]: Looking at the design system, these route declarations usually live in a `routes.js`. I quite like that separation. But I don't know if that's an Express convention or a Design System convention

File: app/assets/scss/app.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+$govuk-show-breakpoints: true;
+
+@import "../../../src/govuk-frontend/all.scss";
```
Comment (benvand) [inline app/assets/scss/app.scss]: Defo can't reference this any other way? Could be fragile

File: app/app.js (hunk 2 above)
Comment (lfdebrux) [reply app/app.js]: I agree that's a nice convention, I also think though that we should try and keep the app javascript as close to what GOV.UK Frontend uses, so we can keep up with any features they add/bugs they fix in future...

**Comment by lfdebrux** [comment]
Looks good to me, just needs README going over once more.

File: app/assets/scss/app.scss (hunk 3 above)
Comment (aliuk2012) [reply app/assets/scss/app.scss]: I was having a look at this and we might be able to simplify this.

File: app/app.js (hunk 2 above)
Comment (aliuk2012) [reply app/app.js]: Just some background info, Express.js is not opinionated which is why there isn't really a set convention if you try to find one. A lot of patterns are copied from other MVC type frameworks.

File: app/assets/scss/app.scss (hunk 3 above)
Comment (aliuk2012) [reply app/assets/scss/app.scss]: Should be fixed now

File: app/app.js (hunk 2 above)
Comment (benvand) [reply app/app.js]: 👍 Fair enough

File: app/assets/scss/app.scss (hunk 3 above)
Comment (benvand) [reply app/assets/scss/app.scss]: Nice 👍

**Pull request merged**